### PR TITLE
#1-4Task. Elizaveta Panasiuk

### PR DESCRIPTION
**Pull request by ElizavetaPanasiuk**
https://travis-ci.com/github/ElizavetaPanasiuk/js-assignments/builds/175854360

### CODE REVIEW

**Comment by ElizavetaPanasiuk** [comment]
Please don't pay attention to commits after "Update task n". Github showed conflicts and after the merge tests in travis show errors and I can't decide this problem.

File: task/04-arrays-tasks.js [hunk 1]
```diff
@@ -519,7 +695,14 @@ function group(array, keySelector, valueSelector) {
  *   ['one','two','three'], x=>x.split('')  =>   ['o','n','e','t','w','o','t','h','r','e','e']
  */
 function selectMany(arr, childrenSelector) {
-    throw new Error('Not implemented');
+   let result = [];
+
+   arr.map(function (elem) {
+      elem = childrenSelector(elem);
```
Comment (ilya4321) [inline task/04-arrays-tasks.js]: you are using the `map` as loop again

File: task/04-arrays-tasks.js [hunk 2]
```diff
@@ -503,7 +665,21 @@ function distinct(arr) {
  *   }
  */
 function group(array, keySelector, valueSelector) {
-   throw new Error('Not implemented');
+   let keys = array.map(keySelector);
+   let values = array.map(valueSelector);
+   let map = new Map();
+
+   keys.filter((elem, index) => {
```
Comment (ilya4321) [inline task/04-arrays-tasks.js]: You are using the `filter` as loop here. Please try reduce

File: task/04-arrays-tasks.js [hunk 3]
```diff
@@ -469,7 +624,14 @@ function getIntervalArray(start, end) {
  *   [ 1, 1, 2, 2, 3, 3, 4, 4] => [ 1, 2, 3, 4]
  */
 function distinct(arr) {
-   throw new Error('Not implemented');
+   let result = [];
+   arr.map(function (elem) {
+      if (result.indexOf(elem) === -1) {
```
Comment (ilya4321) [inline task/04-arrays-tasks.js]: you are using the `map` as loop here. That's a hack for the task ) please try with reduce instead

**Review by IlyaMokin — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by IlyaMokin** [comment]
Thanks, closing as accepted!

**Pull request closed without merging**